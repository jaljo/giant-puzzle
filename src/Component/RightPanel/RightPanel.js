import React from 'react'
import './RightPanel.scss'
import chickFace from "./../../Assets/images/chick_face.png"
import foxFace from "./../../Assets/images/fox_face.png"
import burrow from "./../../Assets/images/burrow.png"
import goal from "./../../Assets/images/goal.png"

// RightPanel :: Props -> React.Component
const RightPanel = ({
  retry,
  gameOver,
  gameWon,
}) =>
  <div data-is="right-panel">
    {(!gameWon && !gameOver) &&
      <>
        <h1>La chasse aux poussins est ouverte !</h1>
        <p>
          Bravo, tu es arrivé jusqu'ici ! Je suis l'<b>agent CC</b> et j'ai une
          mauvaise nouvelle pour toi, tu t'es transformé en poussin&nbsp;
          <img src={chickFace} alt="chick"/> !
        </p>
        <p>
          Fais tomber les renards&nbsp;
          <img src={foxFace} alt="fox" /> dans leurs terriers&nbsp;
          <img src={burrow} alt="burrow" /> grâce aux flèches de
          ton clavier.
        </p>
        <p>
          Pour réussir, les deux renards&nbsp;
          <img src={foxFace} alt="fox" /> doivent être dans leurs
          terriers <img src={goal} alt="goal" /> en même temps !
        </p>
        <p>Prends bien garde à ne pas te faire manger ou tu devras recommencer !</p>
      </>
    }

    {gameWon &&
      <>
        <p className="emphasis">Félicitations !</p>
        <p>
          Tu as réussi à sauver ta peau ! Tu es le futur&nbsp;
          <b><i>Sherlock Holmes</i></b> de notre équipe !
        </p>
      </>
    }

    {gameOver &&
      <p className="emphasis">
        GAME OVER !
      </p>
    }

    <button className="retry" onClick={retry}>
      Recommencer !
    </button>
  </div>

export default RightPanel
