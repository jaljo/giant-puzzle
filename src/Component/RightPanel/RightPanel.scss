@import './../../Style/variables';

div[data-is="right-panel"] {
  padding: 0 1rem;

  h1 {
    margin-top: 0;
  }

  p {
    font-size: 1rem;

    &.emphasis {
      margin: 0;
      text-align: center;
      color: $color-white;
      font-size: 3rem;
      font-weight: bold;
      text-shadow: 0px 3px 0px $color-soil-light;
    }

    img {
      max-height: 30px;
    }
  }

  button.retry {
    display: block;
    margin: 3rem auto 0 auto;
    padding: 1rem;
    min-width: 200px;
    font-size: 0.9rem;
    font-weight: bold;
    color: $color-white;
    background-color: $color-grass-dark;
    box-shadow: 0px 3px 0px $color-soil-dark;
    border: none;
    outline: none;
    cursor: pointer;

    &:hover {
      background-color: $color-grass-light;
      box-shadow: 0px 3px 0px $color-soil-light;
    }

    &:active {
      box-shadow: inset 0px 3px $color-soil-light;
    }

    &::-moz-focus-inner {
      border: 0;
    }
  }
}
