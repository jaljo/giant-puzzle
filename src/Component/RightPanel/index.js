import RightPanel from './RightPanel'
import { connect } from 'react-redux'
import { compose } from 'ramda'
import { retry } from './../../Redux/State/Board'

// mapStateToProps :: State -> Props
const mapStateToProps = state => ({
  gameOver: state.Board.gameOver,
  gameWon: state.Board.winGame,
})

// mapDispatchToProps :: (Action * -> State) -> Props
const mapDispatchToProps = dispatch => ({
  retry: compose(dispatch, retry),
})

// RightPanel :: Props -> React.Component
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(RightPanel)
