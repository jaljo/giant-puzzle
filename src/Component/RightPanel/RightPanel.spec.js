import RightPanel from './RightPanel'
import React from 'react'
import renderer from 'react-test-renderer'

describe('Component/RightPanel', () => {
  it('renders the welcome message when the game is neither won or lost', () => {
    const state = {
      gameWon: false,
      gameOver: false,
    }

    expect(
      renderer.create(<RightPanel {...state} />)
    ).toMatchSnapshot()
  })

  it('renders the win message when the game is won', () => {
    const state = {
      gameWon: true,
    }

    expect(
      renderer.create(<RightPanel {...state} />)
    ).toMatchSnapshot()
  })

  it('renders the game over message when the game is lost', () => {
    const state = {
      gameOver: true,
    }

    expect(
      renderer.create(<RightPanel {...state} />)
    ).toMatchSnapshot()
  })
})
