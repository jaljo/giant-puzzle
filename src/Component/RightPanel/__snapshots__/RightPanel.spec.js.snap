// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component/RightPanel renders the game over message when the game is lost 1`] = `
<div
  data-is="right-panel"
>
  <p
    className="emphasis"
  >
    GAME OVER !
  </p>
  <button
    className="retry"
  >
    Recommencer !
  </button>
</div>
`;

exports[`Component/RightPanel renders the welcome message when the game is neither won or lost 1`] = `
<div
  data-is="right-panel"
>
  <h1>
    La chasse aux poussins est ouverte !
  </h1>
  <p>
    Bravo, tu es arrivé jusqu'ici ! Je suis l'
    <b>
      agent CC
    </b>
     et j'ai une mauvaise nouvelle pour toi, tu t'es transformé en poussin 
    <img
      alt="chick"
      src="/images/chick_face.png"
    />
     !
  </p>
  <p>
    Fais tomber les renards 
    <img
      alt="fox"
      src="/images/fox_face.png"
    />
     dans leurs terriers 
    <img
      alt="burrow"
      src="/images/burrow.png"
    />
     grâce aux flèches de ton clavier.
  </p>
  <p>
    Pour réussir, les deux renards 
    <img
      alt="fox"
      src="/images/fox_face.png"
    />
     doivent être dans leurs terriers 
    <img
      alt="goal"
      src="/images/goal.png"
    />
     en même temps !
  </p>
  <p>
    Prends bien garde à ne pas te faire manger ou tu devras recommencer !
  </p>
  <button
    className="retry"
  >
    Recommencer !
  </button>
</div>
`;

exports[`Component/RightPanel renders the win message when the game is won 1`] = `
<div
  data-is="right-panel"
>
  <p
    className="emphasis"
  >
    Félicitations !
  </p>
  <p>
    Tu as réussi à sauver ta peau ! Tu es le futur 
    <b>
      <i>
        Sherlock Holmes
      </i>
    </b>
     de notre équipe !
  </p>
  <button
    className="retry"
  >
    Recommencer !
  </button>
</div>
`;
